src/routes/safe/components/Apps/hooks/useAppList.ts

```diff
@@ -130,9 +71,99 @@ const useAppList = (): UseAppListReturnType => {
     [appList],
   )
 
+  // Load apps list
+  // for each URL we return a mocked safe-app with a loading status
+  // it was developed to speed up initial page load, otherwise the
+  // app renders a loading until all the safe-apps are fetched.
+  useEffect(() => {
+    const loadApps = async () => {
+      // recover apps from storage:
+      // * third-party apps added by the user
+      // * disabled status for both static and third-party apps
+      const persistedAppList = (await loadFromStorage<StoredSafeApp[]>(APPS_STORAGE_KEY)) || []
+      let list: (StoredSafeApp & { isDeletable: boolean; networks?: number[] })[] = persistedAppList.map((a) => ({
+        ...a,
+        isDeletable: true,
+      }))
+
+      // merge stored apps with static apps (apps added manually can be deleted by the user)
+      staticAppsList.forEach((staticApp) => {
+        const app = list.find((persistedApp) => persistedApp.url === staticApp.url)
+        if (app) {
+          app.isDeletable = false
+          app.networks = staticApp.networks
+        } else {
+          list.push({ ...staticApp, isDeletable: false })
+        }
+      })
+
+      // filter app by network
+      list = list.filter((app) => {
+        // if the app does not expose supported networks, include them. (backward compatible)
+        if (!app.networks) {
+          return true
+        }
+        return app.networks.includes(getNetworkId())
+      })
+
+      const apps: SafeApp[] = []
+      for (let index = 0; index < list.length; index++) {
+        const currentApp = list[index]
+        const appUrl = currentApp.url.trim()
+
+        const appInfo = {
+          ...getEmptySafeApp(),
+          url: appUrl,
+          name: 'loading',
+          iconUrl: loaderDotsSvg,
+        }
+
+        appInfo.disabled = Boolean(currentApp.disabled)
+        appInfo.isDeletable = Boolean(currentApp.isDeletable) === undefined ? true : currentApp.isDeletable
+
+        apps.push(appInfo)
+      }
+
+      setAppList(apps)
+    }
+
+    if (!appList.length) {
+      loadApps()
+    }
+  }, [appList])
```

Comment:
Seems like it will end up in an infinite loop

Reply:
no really, it will run only once, if appList is empty.

Reply:
ah didn't spot the if statement